src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp

```diff
@@ -1011,14 +1011,36 @@ DQLiftGatherSymCW::DQLiftGatherSymCW() {
         auto new_cvt_w = std::make_shared<ov::op::v0::Convert>(matched_out_w, ov::element::f16);
         auto gather_c = std::make_shared<ov::op::v0::Constant>(ov::element::i32, ov::Shape{}, 0);
         auto new_g_w = std::make_shared<ov::op::v8::Gather>(new_cvt_w, matched_out_ids, gather_c);
-        auto new_g_s = std::make_shared<ov::op::v8::Gather>(matched_out_s, matched_out_ids, gather_c);
+        auto new_scale_out = matched_out_s;
+
+        auto qcvtm_iter = node_to_output.find(qcvtm);
+        if (qcvtm_iter == node_to_output.end()) {
+            new_scale_out = std::make_shared<ov::op::v0::Convert>(matched_out_s, ov::element::f16);
+        }
```

Comment:
What's the idea to add a convert here? I believe we just lifting the gathers here but not lowering the stuff down to FP16 here, shouldn't the HostGather pass do that instead?

Reply:
Previously we had weights being converted to fp16 above and scale in fp16. Now we have scale in fp32, so we need to match weight and scale precision. Since we are updating graph here, it's not possible to do it in HostGather

Reply:
Changed the check to check scale's precision